Modifed YAML and SQL CI

MartyC-137 opened the pull request:
Added the following code to the final bash scripts to minimize piping issues:

```bash
changed = git diff-tree --no-commit-id --name-only -r $(Build.SourceVersion) | grep '\.sql$'

          if [[ "${#changed[@]}" -gt 0 ]]; then
            for filename in "${#changed[@]}"; do
              if [[ -f "$filename" ]]; then
                echo "linting $filename" && sqlfluff lint "$filename" --dialect snowflake
            else
              echo "File not found: $filename"
            fi
          done
```

MartyC-137 committed:
Modifed YAML and SQL CI per recommendations

nidan841g commented:
You're missing some parens there. That should be

```bash
changed=( $(git diff-tree --no-commit-id --name-only -r $(Build.SourceVersion) | grep '\.sql$') )
```
The outer pair is for the array the inner `$(command ...)` is a [command substitution](https://www.gnu.org/software/bash/manual/html_node/Command-Substitution.html).

nidan841g commented:
Also possibly need a `fi` to match the first `if [[ "${#changed[@]}" -gt 0 ]]; then`

MartyC-137 committed:
Minor updates to sql and yaml linter ci

MartyC-137 commented:
Updated code:

```bash
changed=( $(git diff-tree --no-commit-id --name-only -r $(Build.SourceVersion) | grep '\.sql$') )

if [[ "${#changed[@]}" -gt 0 ]]; then
  for filename in "${#changed[@]}"; do
    if [[ -f "$filename" ]]; then
      echo "linting $filename" && sqlfluff lint "$filename" --dialect snowflake
    else
      echo "File not found: $filename"
    fi
  done
fi
```

nidan841g commented:
You want `for filename in "${changed[@]}"; do` (lose the # char in there)

Commit:
Updated for loop, removed #

MartyC-137 commented:
> You want `for filename in "${changed[@]}"; do` (lose the # char in there)

Fixed
